src/graphics.rs

```diff
@@ -279,6 +279,70 @@ pub struct BlendState {
     pub dst_alpha: BlendFactor,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct StencilState {
+    pub front: StencilFaceState,
+    pub back: StencilFaceState,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct StencilFaceState {
+    // gl.stencilOpSeparate(face, fail_op, depth_fail_op, pass_op)
+    /// Operation to use when stencil test fails
+    pub fail_op: StencilOp,
+    
+    /// Operation to use when stencil test passes, but depth test fails
+    pub depth_fail_op: StencilOp,
+    
+    /// Operation to use when both stencil and depth test pass, 
+    /// or when stencil pass and no depth or depth disabled
+    pub pass_op: StencilOp,
+    
+    // gl.stencilFuncSeparate(face, test_func, test_ref, test_mask);
```

Comment:
Taking future multiple backends into account it will not always be `gl.*`, the actual implementation will depend on the backend.
So I would try to avoid references to GL in the comments to avoid frustration and leave the room to change the implementation.

Reply:
fixed